instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -346,6 +348,11 @@ def _start_response(status, response_headers, *args, **kwargs):
                 excluded_urls is None
                 or not excluded_urls.url_disabled(flask.request.url)
             ):
+                if flask.request.url_rule:
+                    wrapped_app_environ[_ENVIRON_REQUEST_ROUTE_KEY] = str(
```

Comment:
Why do we need to store this value in the environ? It looks like it's only being used in `_start_response`?

Reply:
I set the value in the `_start_response` where I can access the variable `flask.request.url_rule` to populate the` http.route` or `http.target`. Otherwise, if I try using this value directly without a variable, I get the error:
`RuntimeError: Working outside of request context.`
This was the motivation for using the variable. For more context, we already discussed this in the first comment of this PR.

Reply:
But you can use a normal Python variable with `nonlocal`, you don't need the environ. That's what I said in my first comment.

Reply:
I change to nonlocal variable in mi last commit. Hope its okey :)! @alexmojaki @lzchen